internal/host/hostcommands/install_cmd.go

```diff
@@ -255,35 +256,42 @@ func (i *installCmd) getDiskUnbootableCmd(ctx context.Context, host models.Host)
 /*
 	This function combines existing InstallerArgs ( set by user for his own reasons ) with the
 	--copy-network argument needed by the static ips configuration. In case user has also
-	set --copy-network, function will set only one such argument
+	set --copy-network, function will set only one such argument. It also append an arg that
+	controls DHCP depending on the IP stack being used.
 */
 func constructHostInstallerArgs(cluster *common.Cluster, host *models.Host) (string, error) {
-	if cluster.ImageInfo.StaticNetworkConfig == "" {
-		return host.InstallerArgs, nil
-	}
 
-	if host.InstallerArgs == "" {
-		newArgs := []string{"--copy-network"}
-		argsBytes, err := json.Marshal(newArgs)
+	var installerArgs []string
+	if host.InstallerArgs != "" {
+		err := json.Unmarshal([]byte(host.InstallerArgs), &installerArgs)
 		if err != nil {
 			return "", err
 		}
-		return string(argsBytes), nil
 	}
 
-	var currentInstallerArgs []string
-	err := json.Unmarshal([]byte(host.InstallerArgs), &currentInstallerArgs)
-	if err != nil {
-		return "", err
+	if cluster.MachineNetworkCidr != "" && !funk.ContainsString(installerArgs, "ip=dhcp6") && !funk.ContainsString(installerArgs, "ip=dhcp") {
```

Comment:
funk.ContainsString(installerArgs, "ip=dhcp") contains also funk.ContainsString(installerArgs, "ip=dhcp6")

Reply:
Hmmm, according to https://github.com/thoas/go-funk/blob/master/typesafe.go#L344, it looks for an exact match.